### PR TITLE
Docs - TypeScript 5.0 support

### PR DESCRIPTION
**Pull request by ben-polinsky**
Use typedoc 0.23 and TS 5 for generating docs.

### CODE REVIEW

File: common/config/rush/pnpm-lock.yaml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 lockfileVersion: 5.3
 
+overrides:
```
Comment (raplemie) [inline common/config/rush/pnpm-lock.yaml:3]: Where are these comming from ? Were they added manually ? Will they get wiped next time we `rush update` ?

Comment (ben-polinsky) [reply common/config/rush/pnpm-lock.yaml:3]: Was a first attempt. They should be gone on update, I can do that and test everything works.

Comment (ben-polinsky) [reply common/config/rush/pnpm-lock.yaml]: Gone.

Comment (ben-polinsky) [reply common/config/rush/pnpm-lock.yaml]: Was an issue with the validation in core... should hopefully be fixed now... let's see

**Review by raplemie — APPROVED** [review]
LGTM

**Comment by raplemie** [comment]
@mergifyio backport release/4.0.x

**Pull request merged**